File: api/app/v1/models/OperationalList.go
package models

type OperationalList struct {
	ID          uint        `json:"id,omitempty" gorm:"primary_key"`
	ReferenceID uint        `json:"reference_id,omitempty"`
	Reference   *Reference  `json:"reference,omitempty"`
	Operations  []Operation `json:"operations,omitempty" gorm:"foreignKey:OperationalListID"`
}

type Operation struct {
	ID                uint             `json:"id,omitempty" gorm:"primary_key"`
	OperationalListID uint             `json:"operational_list_id,omitempty"`
	OperationalList   *OperationalList `json:"operational_list,omitempty"`
	Description       string           `json:"description,omitempty"`
}

File: client/src/schema/operationalList.ts
import type { MergeCoreCollection, VytexReference } from '../index.js';

export type VytexOperationalList<Schema = any> = MergeCoreCollection<
	Schema,
	'vytex_operational_list',
	{
		id: number;
		reference_id: number | null;
		reference: VytexReference<Schema> | null;
		operations: Operation<Schema>[] | null;
	}
>;

export type Operation<Schema = any> = MergeCoreCollection<
	Schema,
	'vytex_operation',
	{
		id: number;
		operational_list_id: number | null;
		operational_list: VytexReference<Schema> | null;
		description: string | null;
	}
>;
